ios/Classes/AudioPlayer.swift

```diff
@@ -69,54 +92,59 @@ class AudioPlayer : NSObject, AVAudioPlayerDelegate {
         startListening()
         result(true)
     }
-    
-    func pausePlayer(result:  @escaping FlutterResult){
+
+    func pausePlayer(result: @escaping FlutterResult) {
         stopListening()
         player?.pause()
         result(true)
     }
-    
-    func stopPlayer(result:  @escaping FlutterResult){
+
+    func stopPlayer(result: @escaping FlutterResult) {
         stopListening()
         player?.stop()
         player = nil
         timer = nil
         result(true)
     }
-    
-    
-    func getDuration(_ type:DurationType,_ result:  @escaping FlutterResult) throws {
+
+
+    func getDuration(_ type: DurationType, _ result: @escaping FlutterResult) throws {
         if type == .Current {
             let ms = (player?.currentTime ?? 0) * 1000
             result(Int(ms))
-        }else{
+        } else {
             let ms = (player?.duration ?? 0) * 1000
             result(Int(ms))
         }
     }
-    
-    func setVolume(_ volume: Double?,_ result : @escaping FlutterResult) {
+
+    func setVolume(_ volume: Double?, _ result: @escaping FlutterResult) {
         player?.volume = Float(volume ?? 1.0)
         result(true)
     }
-    
-    func seekTo(_ time: Int?,_ result : @escaping FlutterResult) {
-        player?.currentTime = Double(time!/1000)
-        result(true)
+
+    func seekTo(_ time: Int?, _ result: @escaping FlutterResult) {
+        if(time != nil) {
+            player?.currentTime = Double(time! / 1000)
+            result(true)
+        } else {
+            result(false)
+        }
+        
     }
-    
-    func startListening(){
+
+    func startListening() {
         if #available(iOS 10.0, *) {
```

Comment:
Remove this condition if possible.

Reply:
This condition is required because Timer.schedules timer is available ios 10 or above